webactions.md

```diff
@@ -0,0 +1,546 @@
+    "__ow_headers": {
+      "accept": "*/*",
+      "connection": "close",
+      "host": "172.17.0.1",
+      "user-agent": "curl/7.43.0"
+    },
+    "__ow_path": ""
+  }
+}
+```
+
+You can use form data as input:
+
+```bash
+$ curl https://${APIHOST}/api/v1/web/guest/demo/hello.json -d "name":"Jane"
+{
+  "response": {
+    "name": "Jane",
+    "__ow_method": "post",
+    "__ow_headers": {
+      "accept": "*/*",
+      "connection": "close",
+      "content-length": "10",
+      "content-type": "application/x-www-form-urlencoded",
+      "host": "172.17.0.1",
+      "user-agent": "curl/7.43.0"
+    },
+    "__ow_path": ""
+  }
+}
+```
+
+You can also invoke the action with a JSON object.
+
+```bash
+$ curl https://${APIHOST}/api/v1/web/my-namespace/demo/hello.json -H 'Content-Type: application/json' -d '{"name":"Jane"}'
+{
+  "response": {
+    "name": "Jane",
+    "__ow_method": "post",
+    "__ow_headers": {
+      "accept": "*/*",
+      "connection": "close",
+      "content-length": "15",
+      "content-type": "application/json",
+      "host": "172.17.0.1",
+      "user-agent": "curl/7.43.0"
+    },
+    "__ow_path": ""
+  }
+}
+```
+
+For convenience, query parameters, form data, and JSON object body entities are all treated as dictionaries, and their values are directly accessible as action input properties. This is not the case for web actions that opt to handle HTTP request entities more directly or when the web action receives an entity that is not a JSON object.
+
+Here is an example of using a "text" content-type with the same example shown above.
+
+```bash
+$ curl https://${APIHOST}/api/v1/web/my-namespace/demo/hello.json -H 'Content-Type: text/plain' -d "Jane"
+{
+  "response": {
+    "__ow_method": "post",
+    "__ow_headers": {
+      "accept": "*/*",
+      "connection": "close",
+      "content-length": "4",
+      "content-type": "text/plain",
+      "host": "172.17.0.1",
+      "user-agent": "curl/7.43.0"
+    },
+    "__ow_path": "",
+    "__ow_body": "Jane"
+  }
+}
+```
+
+## Content extensions
+
+A content extension is generally required when invoking a web action; the absence of an extension assumes `.http` as the default. The fully qualified name of the action must include its package name, which is `default` if the action is not in a named package.
+
+
+## Protected parameters
+
+Action parameters are protected and treated as immutable. Parameters are automatically finalized when enabling web actions.
+
+```bash
+$ nim action create /my-namespace/demo/hello hello.js \
+      --parameter name Jane \
+      --web true
+```
+
+The result of these changes is that the `name` is bound to `Jane` and cannot be overridden by query or body parameters because of the final annotation. This secures the action against query or body parameters that try to change this value whether by accident or intentionally.
+
+## Securing web actions
+
+**[[NH: I don't know what to do with this section, please advise.]]**
```

Comment:
Mostly, the section is still applicable substituting `nim` for `wsk`.   I'll note exceptions.